runtime/src/chpl-topo.c

```diff
@@ -37,6 +38,9 @@
 #include <string.h>
 #include <unistd.h>
 
+#if defined(CHPL_HWLOC_HWLOC) || defined(CHPL_HWLOC_SYSTEM)
+#define CHPL_HAS_HWLOC
```

Comment:
I generally prefer this style.
1) It enables more complicated ifdef logic in the future / separates the concern of deciding what it means to have hwloc
2) It becomes possible for a user to set -DCHPL_HAS_HWLOC (with our direction) in the event there is something wrong with the auto-detection